### PR TITLE
Issues/kp 367 using ladok for search courses

### PR DESCRIPTION
**Pull request by amirhossein-haerian**
This PR is dependent to -> https://github.com/KTH/studadm-om-kursen-packages/pull/11

Things to mention:

1. We are getting fewer results from Ladok API
2. Filter filters only work for the first selected value as Ladok API does not currently support multiple selected searchParams.
3. We don't have any searhcParam for "Behandlar miljö, miljöteknik eller hållbar utveckling" so it will be implemented after we get the functionality inside the Ladok API.
4. The "Nedlagd kurs" filter is now implemented using "avvecklad" searchParam inside the Ladok API but it seems that this is not exactly what we want.
5. Right now searchParams are not working when we select all of the departments inside one school.

### CODE REVIEW

**Review by axelbjo — APPROVED** [review]
Looks good to me, great job! One small comment/question though

File: public/js/app/hooks/useCourseSearch.ts
[the remark this answers is not in the document]
Comment (amirhossein-haerian) [reply public/js/app/hooks/useCourseSearch.ts:1]: At some point, I thought changing the condition from "OR" to "AND" would be more beneficial and would make sense more, but I have not seen improvement during testing and I forgot to revert it. So I revert it now.

Thanks for mentioning it.

**Comment by karlandindrakryggen** [comment]
There is no end period here for some reason, just "P" (´P1 HT24 - P HT24`).  Maybe fallback to just show the start period `P1 HT24` is the endperiod calculation fails? 

image

**Comment by karlandindrakryggen** [comment]
I think we need better error handling/fallback in `<SearchAlert>`.  

In my case i first forgot to add correct client secret env variable and got `Error: no accesstoken` and the react code crashed at this line `const { header, help, text } = searchAlarms[externalAlertType]` when externalAlertType was `ERROR-courseSearch-500`. Would be good with some generic error message in the alert if it is and unexpected error like this one.

**Comment by karlandindrakryggen** [comment]
I think we have the opportunity to impose the case of overflow. Wouldn't it be possible to actually show the first results we find and replace the text this alert with something like "There are too many courses that match your search critera. **We are showing the first NN results. [...]**. 

But I think it could good to do this improvement in a separate task/PR after this is merged.

image

**Comment by amirhossein-haerian** [comment]
> I think we need better error handling/fallback in `<SearchAlert>`.
> 
> In my case i first forgot to add correct client secret env variable and got `Error: no accesstoken` and the react code crashed at this line `const { header, help, text } = searchAlarms[externalAlertType]` when externalAlertType was `ERROR-courseSearch-500`. Would be good with some generic error message in the alert if it is and unexpected error like this one.

I agree with you, a default value for searchAlert is added now.

File: public/js/app/components/mocks/mockCourseSeasrch.ts
Comment (karlandindrakryggen) [inline public/js/app/components/mocks/mockCourseSeasrch.ts:1]: mockCourseSeasrch -> mockCourseSearch

**Review by karlandindrakryggen — APPROVED** [review]
I think this is a really good start! I've added som comments to the [lib PR](https://github.com/KTH/studadm-om-kursen-packages/pull/11). 

A good next step is to remove the old search from this branch.

**Pull request merged**